include/svs/lib/threads/threadpool.h

```diff
@@ -255,6 +255,31 @@ auto create_on_nodes(InterNUMAThreadPool& threadpool, F&& f)
 }
 #endif
 
+/////
+///// A thread pool that dynamically switches between single-threaded and multi-threaded execution.
+///// - If `n == 1`, the task will be executed on the main thread without any locking mechanism.
+///// - For `n > 1`, the tasks will be delegated to the internal `NativeThreadPool` for parallel execution.
+/////
+class SwitchNativeThreadPool {
```

Comment:
According to the [`NativeThreadPoolBase ::parallel_for()`](https://github.com/intel/ScalableVectorSearch/blob/main/include/svs/lib/threads/threadpool.h#L189) implementation, a task is executed in the current thread in case of `n == 1`.
Why we cannot just fix existing code rather than making one more class?

Reply:
Ok, thanks. Looking to `NativeThreadPool::size()` code I did not realize that `NativeThreadPool` is thread-safe.